pkg/reconciler/buildrun/controller.go

```diff
@@ -77,18 +78,40 @@ func add(mgr manager.Manager, r reconcile.Reconciler, maxConcurrentReconciles in
 			o := e.ObjectOld
 			n := e.ObjectNew
 
-			// Process an update event when the old TR resource is not yet started and the new TR resource got a
-			// condition of the type Succeeded
-			if o.Status.StartTime.IsZero() && n.Status.GetCondition(apis.ConditionSucceeded) != nil {
+			// Check for completion time changes
+			if o.Status.CompletionTime == nil && n.Status.CompletionTime != nil {
 				return true
 			}
```

Comment:
Why do we need to reconcile the BuildRun here again when the CompletionTime appeared ?